src/aspire/operators/diag_matrix.py

```diff
@@ -470,11 +470,11 @@ def apply(self, X):
         Define the apply option of a diagonal matrix with a matrix of
         coefficient vectors.
 
-        :param X: Coefficient matrix, each column is a coefficient vector.
+        :param X: Coefficient matrix (ndarray), each column is a coefficient vector.
         :return: A matrix with new coefficient vectors.
         """
 
-        return self * DiagMatrix(X)
+        return (self * DiagMatrix(X.T)).asnumpy().T
```

Comment:
Yeah, I guess I was confused because that line in the docstring was already there. Wasn't sure if this was bug fix or changing the intend behavior of the function.

All good!